batavia/types/Slice.js

```diff
@@ -4,7 +4,7 @@
 
 batavia.types.Slice = function() {
     function Slice(kwargs) {
-        this.start = kwargs.start || 0;
+        this.start = kwargs.start;
```

Comment:
Old code assumed a default of `0` for a slices start value, but when you use a negative step the default value for start is actually the length of the list/str/etc. So it makes more sense to allow a default of null (same as stop) and to handle setting the default in the slice implementation.